### PR TITLE
Feature/nathantypanski/commands

### PR DESCRIPTION
**Pull request by nathantypanski**
Everything up until "echo". Basic string library.

### CODE REVIEW

File: kernel/cstr.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            }
+            s.add_u8(self.get_char(ind) as u8);
+            ind += 1;
+            if (ind == self.size) {
+                return s;
+            }
+        }
+    }
+    
+    #[allow(dead_code)]
```
Comment (lenary) [inline kernel/cstr.rs]: do we get a warning if we get rid of this declaration?

File: kernel/memory/allocator.rs [hunk 2]
```diff
@@ -194,10 +195,13 @@ impl Allocator for Alloc {
...
-            return (
-                mut_offset(self.base, (offset << self.el_size) as int),
-                size << self.el_size
-            )
+            let roffset = mut_offset(self.base, (offset << self.el_size) as int);
+            let rsize = size << self.el_size;
+            //sgash::putchar('o');
+            //sgash::putnum(roffset as uint, 0);
+            //sgash::putchar('s');
+            //sgash::putnum(rsize as uint, 0);
```
Comment (lenary) [inline kernel/memory/allocator.rs]: these really are dead bits of code. Just delete them :)

**Comment by lenary** [comment]
Address those comments then I'll merge (I'll probably [squash](http://git-scm.com/book/ch6-4.html#Squashing-Commits) in the merge to have fewer commits in the history, but that shouldn't be an issue)

**Comment by nathantypanski** [comment]
Fixed in 998528d.

**Comment by lenary** [comment]
This all looks good. I'll hit merge.

**Comment by lenary** [comment]
in fact, merge when you're ready.

**Pull request merged**